### PR TITLE
Pass session_id during Websocket connect

### PR DESCRIPTION
**Pull request by gogasca**
Fixes: https://github.com/jupyter-server/jupyter_server/issues/1439

Pass HTTP query param session_id during Websocket channel connection.
Example:

```
GET /api/kernels/342eba8f-aa8c-47a9-8fee-8541cee8ddfe/channels
```

vs

```
GET /api/kernels/342eba8f-aa8c-47a9-8fee-8541cee8ddfe/channels?session_id=4feb3f46-4e83-4fba-bebd-3aba95e96fc0 
```

### CODE REVIEW

File: jupyter_server/gateway/connections.py [hunk 1]
```diff
@@ -47,6 +47,8 @@ async def connect(self):
             url_escape(self.kernel_id),
             "channels",
         )
+        if self.session_id:
+            ws_url += f"?session_id={self.session_id}"
```
Comment (krassowski) [inline jupyter_server/gateway/connections.py:51]: Should `session_id` be escaped with `escape_url`? Also a test case would be nice :)

Comment (gogasca) [reply jupyter_server/gateway/connections.py]: used url_escape + add a test

Comment (gogasca) [reply jupyter_server/gateway/connections.py]: Added a test, but looks like Windows tests are failing for unrelated reason?

Comment (blink1073) [reply jupyter_server/gateway/connections.py]: The windows tests were fixed in #1441

Comment (gogasca) [reply jupyter_server/gateway/connections.py]: Thanks @blink1073 I synced to main and Windows tests work now, but now I get some coverage error. https://github.com/jupyter-server/jupyter_server/actions/runs/9821413970/job/27117179741?pr=1440 any pointers? Thanks

**Comment by blink1073** [comment]
I kicked the coverage test, it looks like it was a temporary connectivity issue.

**Comment by gogasca** [comment]
krassowski@ would you mind taking another look? Thanks!

**Review by krassowski — COMMENTED** [review]
Looks good to me. It might be worth bringing it up on the [server and kernels meeting](https://github.com/jupyter-server/team-compass?tab=readme-ov-file#weekly-team-meetings) tomorrow to check for other ideas.

**Comment by gogasca** [comment]
@krassowski is it ok to merge? Thanks!

**Comment by krassowski** [comment]
It looks like @Zsailer commented during the meeting, maybe he can approve and merge (I do not have merge rights here).

**Review by Zsailer — APPROVED** [review]
LGTM. Thanks @gogasca!

Pinging @lresende here too, since he's most familiar with the gateways. Luciano, do you know if e.g. enterprise_gateway tracks individual kernel (ZMQ connection) sessions in any way? 

Note, this is different than "sessions" from the server's sessions API, which maps documents to kernels. 

The sessions we are dealing with in this PR are tracking ZMQ client sessions. This flows a unique session_id from the browser-->server-->kernel's session client (i.e. at the ZQM interface). I think (?) that JupyterLab uses this type of session_id to distinguish messages coming from the same kernel to different documents, but I'm not totally sure. 

We use the word "session" in too many places 😅

**Comment by gogasca** [comment]
@Zsailer thanks! do you know when is the next `jupyter_server` release? Thanks

**Comment by Zsailer** [comment]
Released 😎 https://github.com/jupyter-server/jupyter_server/releases/tag/v2.14.2

**Pull request merged**